### PR TITLE
ENH: fallback to stock sklearn for Kmeans CSR inputs in train

### PR DESCRIPTION
**Pull request by samir-nasibli**
### Description

Proposed Second option for 2025.0 release in case if https://github.com/oneapi-src/oneDAL/pull/2906 will not full or mostly fix timeouts issue of Kmeans train with sparse inputs

Workaround for the timeouts, alternative for the deselection on #2059 

Note:
If https://github.com/oneapi-src/oneDAL/pull/2906 will succeed and resolve timout issues, will be preferred and 
current #2061 and  #2059 PRs will be closed.

---

Checklist to comply with before moving PR from draft:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.  
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.

**Performance**

- [ ] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [ ] I have provided justification why performance has changed or why changes are not expected.

### CODE REVIEW

**Comment by ethanglaser** [comment]
Covered by https://github.com/oneapi-src/oneDAL/pull/2906
